tests/Backend/SOX/BranchStorageMetadataTest.php

```diff
@@ -0,0 +1,536 @@
+    private int $defaultBranchId;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+        $developerClient = $this->getDeveloperStorageApiClient();
+        $this->branches = new DevBranches($developerClient);
+        $this->cleanupTestBranches($developerClient);
+    }
+
+    public function testCrossBranchStorageMetadata(): void
+    {
+        [$privilegedClient, $productionTableId, $branchClient, $devTableId] = $this->initResources();
+
+        $prodTable = $privilegedClient->getTable($productionTableId);
+        $bucketStringId = $prodTable['bucket']['id'];
+
+        $branchMetadataApi = new Metadata($branchClient);
+        $productionMetadataApi = new Metadata($privilegedClient);
+
+        assert($productionTableId === $devTableId);
+
+        $bucketDevMetadata = [
+            [
+                'key' => 'branch',
+                'value' => 'dev',
+            ],
+        ];
+        $tableDevMetadata = [
+            [
+                'key' => 'branch',
+                'value' => 'dev',
+            ],
+        ];
+        $columnsDevMetadata = [
+            'name' => [
+                [
+                    'key' => 'branch',
+                    'value' => 'dev',
+                ],
+            ],
+        ];
+
+        // set metadata on branch bucket
+        $branchMetadataApi->postBucketMetadata($bucketStringId, 'test', $bucketDevMetadata);
+
+        // set metadata on branch table
+        $branchMetadataApi->postTableMetadataWithColumns(new TableMetadataUpdateOptions(
+            $devTableId,
+            'test',
+            $tableDevMetadata,
+            $columnsDevMetadata
+        ));
+        $branchMetadataApi->postColumnMetadata($devTableId . '.name', 'test', [
+            [
+                'key' => 'branch-from-column-call',
+                'value' => 'dev',
+            ],
+        ]);
+
+        // production resources does not have metadata
+        $this->assertSame([], $productionMetadataApi->listBucketMetadata($bucketStringId));
+        $this->assertSame([], $productionMetadataApi->listTableMetadata($productionTableId));
+        $this->assertSame([], $productionMetadataApi->listColumnMetadata($productionTableId . '.name'));
+
+        // branch resources have metadata
+        $actualBranchBucketMetadata = $branchMetadataApi->listBucketMetadata($bucketStringId);
+        $this->assertMetadataEquals([
+            'key' => 'branch',
+            'value' => 'dev',
+            'provider' => 'test',
+        ], $actualBranchBucketMetadata[0]);
+        $actualBranchTableMetadata = $branchMetadataApi->listTableMetadata($devTableId);
+        $this->assertMetadataEquals([
+            'key' => 'branch',
+            'value' => 'dev',
+            'provider' => 'test',
+        ], $actualBranchTableMetadata[0]);
+        $this->assertCount(1, $actualBranchTableMetadata);
+        $actualBranchColumnMetadata = $branchMetadataApi->listColumnMetadata($productionTableId . '.name');
```

Comment:
```suggestion
        $actualBranchColumnMetadata = $branchMetadataApi->listColumnMetadata($devTableId . '.name');
```

Reply:
mozna by mi spis davalo smysl se k tomu chovat jako k bucketu - proste jen tableId - to da najevo, ze se ta tabulka musi chovat stejne